xbmc/filesystem/NFSFile.cpp

```diff
@@ -45,16 +45,16 @@ using namespace std::chrono_literals;
 
 namespace
 {
+// Default "lease_time" on most Linux NFSv4 servers are 90s.
```

Comment:
Because this value can be changed is why I avoided adjusting the timeouts in the first place.

Reply:
But Linux default (nfs-kernel-server) is 90s. This is so....

6 minutes does NOT work. If you want to use 6 minutes because "avoid change here" then you need submit patch to Linux kernel

Reply:
Wouldn't an advanced setting for this purpose be better (while keeping the sane defaults)?
I assume there's no easy way to get those values from the server

Reply:
> I assume there's no easy way to get those values from the server

In fact, the spec says that clients should get the lease_time from the server:

> The client queries the server's "lease_time" attribute. (Default 90s on Linux.)

https://linux-nfs.org/wiki/index.php/NFS_lock_recovery_notes

But we are limited to what `libnfs` offers and I haven't found a way to get this (note that this is "new" in NFSv4 and v4 support in libnfs is very limited and NFSv4.1 is unsupported).

Anyway 45s keep alive should work in all cases because common values are 90s (default), 120s or 60s.

30s or less is discouraged.

Reply:
Maybe @sahlberg know more about the leasing period. It would be nice if we could just query it.

Reply:
I also think that it is necessary to clarify here that the keep alive that we have implemented (it was already in NFSv3) are not to keep the NFS server's session active (this concept simply did not exist in NFSv3) but rather to prevent the hard disk from stopping spin up and go into hibernation when the video is paused for several minutes.

It is true that now with NFSv4 the same keep alive works for both.

The keep alive for NFS session does not need to read data from the hard drive, it just sends a RENEW command. On the other hand, to keep the hard drive spinning, it is necessary to read something...